tower-http/src/trace/on_request.rs

```diff
@@ -1,26 +1,30 @@
 use super::DEFAULT_MESSAGE_LEVEL;
 use http::Request;
 use tracing::Level;
+use tracing::Span;
 
 /// Trait used to tell [`Trace`] what to do when a request is received.
 ///
 /// [`Trace`]: super::Trace
 pub trait OnRequest<B> {
     /// Do the thing.
-    fn on_request(&mut self, request: &Request<B>);
+    ///
+    /// `span` can be used to record field values that weren't known when the span was
+    /// created.
+    fn on_request(&mut self, request: &Request<B>, span: &Span);
 }
 
 impl<B> OnRequest<B> for () {
     #[inline]
-    fn on_request(&mut self, _: &Request<B>) {}
+    fn on_request(&mut self, _: &Request<B>, _span: &Span) {}
```

Comment:
would probably just use `_` here, since we don't care about when the reference is dropped:
```suggestion
    fn on_request(&mut self, _: &Request<B>, _: &Span) {}
```